salt/config/__init__.py

```diff
@@ -3336,7 +3336,7 @@ def get_cloud_config_value(name, vm_, opts, default=None, search_global=True):
         if isinstance(vm_[name], types.GeneratorType):
             value = next(vm_[name], '')
         else:
-            if isinstance(value, dict):
+            if isinstance(value, dict) and isinstance(vm_['name'], dict):
```

Comment:
name should not be quoted, it is a variable.

Reply:
That is my bad.

Reply:
My bad too, doh!